### PR TITLE
Adding the doc format for AdaDelta, AdaMax, Adam, AdaGrad, BatchNorm, Clip, Cast and AUC

### PR DESCRIPTION
**Pull request by kavyasrinet**
Adding the proposed documentation format for AdaDelta.
Let's follow the same going forward for all operators.
The operators in this PR are:
- AdaDelta
- AdaMax
- Adam
- AdaGrad
- BatchNorm
- Clip_op
- Cast_op
- AUC_op

### CODE REVIEW

File: paddle/operators/adadelta_op.cc [hunk 1]
```diff
@@ -84,22 +83,21 @@ class AdadeltaOpMaker : public framework::OpProtoAndCheckerMaker {
                    "numerical stability")
         .SetDefault(1.0e-6f);
     AddComment(R"DOC(
-Adadelta Updates Operator.
+Adadelta Optimizer.
 
-This implements the Adadelta optimizer[1]. Adadelta is a per-dimension
-adaptive learning rate method for gradient descent.
+We implement the Adadelta optimizer as explained in:
```
Comment (abhinavarora) [inline paddle/operators/adadelta_op.cc]: Do you think we should use the first person here?

Comment (kavyasrinet) [reply paddle/operators/adadelta_op.cc]: Fixed. Thanks for the comment.

File: paddle/operators/cast_op.cc [hunk 2]
```diff
@@ -23,13 +23,17 @@ class CastOpProtoMaker : public framework::OpProtoAndCheckerMaker {
...
-    AddOutput("Out", "the output tensor of cast op");
-    AddComment(R"DOC(Cast operator.
-cast the input tensor to other data type.
-)DOC");
+    AddInput("X", "The input tensor of cast op");
+    AddOutput("Out", "The output tensor of cast op");
     AddAttr<int>("out_data_type", "output data type");
     AddAttr<int>("in_data_type", "input data type");
+    AddComment(R"DOC(
+Cast operator.
```
Comment (kexinzhao) [inline paddle/operators/cast_op.cc]: operator -> Operator

Comment (kavyasrinet) [reply paddle/operators/cast_op.cc]: Done. Thanks for pointing that out.

**Pull request merged**